src/blackbox/reader.go

```diff
@@ -50,12 +51,17 @@ func (f *FlightLogReader) LoadFile(file io.Reader, ctx context.Context) (<-chan
 			errChan <- err
 			return
 		}
+		// collect stats when the process is done
+		defer func() {
+			f.Stats = frameReader.Stats
+		}()
 
+	Loop:
 		for !frameReader.Finished {
 			// check on every iteration if the context has been canceled and stop processing if so
 			select {
 			case <-ctx.Done():
-				return
+				break Loop
```

Comment:
`f.Stats = frameReader.Stats` would also be called with a return right ? No need to name the loop then